File: LeetCodeRecord/src/main/java/Q0234.java
import java.util.ArrayList;
import java.util.List;

/*
234.回文链表
 */
public class Q0234 {
    public boolean isPalindrome(ListNode head) {
        List<Integer> list = new ArrayList<>();
        while(head!=null){
            list.add(head.val);
            head=head.next;
        }

        int len = list.size();
        for(int i=0;i<=len/2;i++){
            if(list.get(i)!=list.get(len-1-i)){
                return false;
            }
        }
        return true;
    }
}

File: LeetCodeRecord/cpp/链表/Q0234.cpp
//
// Created by Administrator on 2026/9/3.
//

//回文链表

struct ListNode {
    int val;
    ListNode *next;
    ListNode(int x) : val(x), next(nullptr) {}
};

using namespace std;

class Solution {
public:
    bool isPalindrome(ListNode* head) {
        vector<int> vals;
        while (head!=nullptr) {
            vals.push_back(head->val);
            head = head->next;
        }
        int left = 0, right = vals.size()-1;
        while (left<right) {
            if (vals[left]!=vals[right]) {
                return false;
            }
            left++;
            right--;
        }
        return true;
    }
};
